integration-tests/test_client.py

```diff
@@ -82,21 +82,37 @@ def test_verify_logrotate_feature(insights_client):
     Ref : https://bugzilla.redhat.com/show_bug.cgi?id=1940267
 
     Test Steps:
-        1-Count the number of log file in beginning
-        2-Rotate logs by running logrotate command on CLI for insights-client
-        3-Verify that log is rotated and new log file is created
+        1 - Perform register and payload operation to ensure both
+            insights-client.log and insights-client-payload.log files have logs
+        2 - Rotate logs by running logrotate command on CLI for insights-client
+        3 - Verify insights-client.log and insights-client-payload.log size is 0B
+        4 - Verify rotated files are created in log dir by comparing number of log files
     """
 
     logrotate_conf_file_path = "/etc/logrotate.d/insights-client"
     logdir = "/var/log/insights-client/"
+    logfile_insights = "/var/log/insights-client/insights-client.log"
+    logfile_payload = "/var/log/insights-client/insights-client-payload.log"
+
     assert os.path.exists(logrotate_conf_file_path), "logrotate is not configured"
     # It is necessary to perform some command using insights-client
     # to populate logs
-    insights_client.register()
-    number_of_log_files = len(os.listdir(logdir))
+    # Save the archive, to be used while upload operation
+    reg_result = insights_client.run("--register", "--keep-archive")
+    assert conftest.loop_until(lambda: insights_client.is_registered)
+
+    archive_name = reg_result.stdout.split()[-1]
```

Comment:
What's the exact output of `insights-client --register --keep-archive`? This line seems to assume the last line contains only the archive name -- if that is the case, I'd add a comment before this line with a sample of the output, so it's easier to know why this text extraction is done this way.